tests/test_agg_constraints.py

```diff
@@ -279,3 +280,38 @@ def test_multipiecewise_constraint(model, flow):
 
     model.step()
     assert_allclose(D.flow, min((flow - 40) * 0.25, 50.0))
+
+
+def test_dynamic_factors(model):
+
+    model.timestepper.end = Timestamp("2016-01-03")
+
+    A = Input(model, "A", max_flow=10.0)
+    B = Input(model, "B", max_flow=10.0)
+    C = Input(model, "C", max_flow=10.0)
+    Z = Output(model, "Z", max_flow=10.0, cost=-10)
+
+    agg = AggregatedNode(model, "agg", [A, B, C])
+    factor1 = DailyProfileParameter(model, np.append(np.array([0.8, 0.3]), np.ones(364)))
+    factor2 = DailyProfileParameter(model, np.append(np.array([0.1, 0.3]), np.ones(364)))
+    factor3 = DailyProfileParameter(model, np.append(np.array([0.1, 0.4]), np.ones(364)))
+
+    agg.factor_parameters = [factor1, factor2, factor3]
```

Comment:
Can we make this `agg.factors = [factor1, factor2, factor3]`. I think external API is the same regardless of whether you are using floats or parameters.

Reply:
I'll change this